### PR TITLE
added memory control for sharpening and more

### PR DESCRIPTION
**Pull request by horst-n**
added useUSM to default settings in Pageimage.php
added useUSM to $config->imageSizerOptions in config.php
added useUSM to ImageSizer.php
added memory checking for all parts related to sharpening
rearranged freeing of memory with imagedestroy

changed value for defaultGamma from 2.0 to 2.2 in config.php
fixed a typo in Imagesizer.php

--------

@ryancramerdesign: Please refer to this forum thread to get more informations: https://processwire.com/talk/topic/12015-memory-limit-error-with-images/#entry111840

In short, there were uncontrolled memory allocations in the unsharpMask function that has potential to crash pages. This is fixed now. Also I added the ability on user level to specify selection of the used sharpening algorithm.

### CODE REVIEW

File: wire/core/ImageSizer.php [hunk 1]
```diff
@@ -443,19 +446,31 @@ public function ___resize($targetWidth, $targetHeight = 0) {
...
+		if(isset($thumb2)) $thumb2 = null;
+
 		// optionally apply sharpening to the final thumb
 		if($this->sharpening && $this->sharpening != 'none') { // @horst
 			if(IMAGETYPE_PNG != $this->imageType || ! $this->hasAlphaChannel()) {
-				// is needed for the USM sharpening function to calculate the best sharpening params
-				$this->usmValue = $this->calculateUSMfactor($targetWidth, $targetHeight);
-				$thumb = $this->imSharpen($thumb, $this->sharpening);
+				// calculate if there is enough memory available to apply the USM algorithm, if enabled
+				$this->useUSM = self::checkMemoryForImage(array(imagesx($thumb), imagesy($thumb), 3), array(imagesx($thumb), imagesy($thumb), 3)) === false ? false : $this->useUSM;
```
Comment (teppokoivula) [inline wire/core/ImageSizer.php]: Out of interest, shouldn't this check $this->useUSM first? If it's false in the first place, there seems to be no reason to perform a memory check, at least in this context :)

Comment (horst-n) [inline wire/core/ImageSizer.php]: Ähhm, simply said: yes! :+1: 
I will change the order.
